# Study 865 — Credit → Equity Lead-Lag — the teardown

The Granger-style predictive-regression Newey-West *t*, the risk-on/off discrimination, the 1,000-draw label-shift placebo, the two-era cut, the costed SPY↔IEF overlay vs buy-and-hold, and the 20-seed synthetic positive control.

In [1]:
R = {'start': '2007-05-01', 'end': '2026-06-30', 'n_daily': 4822, 'n_weekly': 1001, 'fingerprint': 'c6ce5a5566a2', 'K': 4, 'n_weeks': 996, 'on_frac': 0.582, 'beta': -0.0546, 'beta_t': -1.7, 'per_sd_bps': -19.6, 'r2': 0.601, 'corr': -0.078, 'beta1': -0.1073, 'per_sd1': -19.5, 't1': -1.58, 'r2_1': 0.592, 'beta2': -0.0117, 'per_sd2': -2.9, 't2': -0.23, 'r2_2': 0.013, 'on_bps': 19.5, 'off_bps': 27.0, 'diff_bps': -7.5, 'disc_t': -0.48, 'welch_t': -0.43, 'n_on': 580, 'n_off': 416, 'placebo_obs': -7.5, 'placebo_mean': -0.2, 'placebo_sd': 16.2, 'placebo_p': 0.663, 'placebo_sigma': -0.4, 'placebo_draws': 1000, 'era_early_beta': -0.0439, 'era_early_t': -1.23, 'era_early_n': 501, 'era_late_beta': -0.091, 'era_late_t': -1.33, 'era_late_n': 495, 'n_switches': 218, 't1_net_sharpe': 0.644, 't1_bh_sharpe': 0.645, 't1_net_cagr': 7.35, 't1_bh_cagr': 10.61, 't1_net_dd': -32.7, 't1_bh_dd': -54.6, 't1_active_bps': -7.6, 't1_active_t': -1.22, 't1_cost_yr': 22.8, 't5_net_sharpe': 0.569, 't5_net_cagr': 6.37, 't5_active_bps': -9.3, 't5_active_t': -1.49, 'null_mean_t': -0.26, 'null_sd_t': 0.75, 'null_fire': 0, 'planted_t': 26.7, 'planted_per_sd': 988, 'planted_r2': 76.0, 'planted_active_t': 6.9, 'planted_net_sharpe': 3.81, 'planted_bh_sharpe': 0.19}

## The signal — does the trailing credit trend LEAD next-week SPY?

Regress `r_SPY[t+1] = a + b·trend_k[t] + u`, trend known at the Friday close of week `t`. Credit leading equity means a **positive, robust** slope.

In [2]:
print(f"4-week (headline): slope {R['beta']:+.4f}  per-sigma {R['per_sd_bps']:+.1f} bps  "
      f"NW(6) t = {R['beta_t']:+.2f}  R2 = {R['r2']:.2f}%%  corr = {R['corr']:+.3f}  (n={R['n_weeks']})")
print(f"1-week cross-check: slope {R['beta1']:+.4f}  per-sigma {R['per_sd1']:+.1f} bps  NW t = {R['t1']:+.2f}  R2 = {R['r2_1']:.2f}%%")
print(f"2-week cross-check: slope {R['beta2']:+.4f}  per-sigma {R['per_sd2']:+.1f} bps  NW t = {R['t2']:+.2f}  R2 = {R['r2_2']:.2f}%%")
print('-> WRONG SIGN at every horizon, |t| < 2 throughout, R2 <= 0.6%% (credit explains ~none of next-week stocks)')

4-week (headline): slope -0.0546  per-sigma -19.6 bps  NW(6) t = -1.70  R2 = 0.60%%  corr = -0.078  (n=996)
1-week cross-check: slope -0.1073  per-sigma -19.5 bps  NW t = -1.58  R2 = 0.59%%
2-week cross-check: slope -0.0117  per-sigma -2.9 bps  NW t = -0.23  R2 = 0.01%%
-> WRONG SIGN at every horizon, |t| < 2 throughout, R2 <= 0.6%% (credit explains ~none of next-week stocks)


## Discrimination — next-week SPY on risk-on (trend>0) vs risk-off weeks

In [3]:
print(f"risk-on  ({R['on_frac']*100:.1f}%%, n={R['n_on']}): next-week SPY {R['on_bps']:+.1f} bps")
print(f"risk-off (       n={R['n_off']}): next-week SPY {R['off_bps']:+.1f} bps")
print(f"difference (on - off): {R['diff_bps']:+.1f} bps/week  NW(6) t = {R['disc_t']:+.2f}  Welch t = {R['welch_t']:+.2f}")
print('-> risk-on weeks earn LESS the following week: conditioning on credit adds no forward edge')

risk-on  (58.2%%, n=580): next-week SPY +19.5 bps
risk-off (       n=416): next-week SPY +27.0 bps
difference (on - off): -7.5 bps/week  NW(6) t = -0.48  Welch t = -0.43
-> risk-on weeks earn LESS the following week: conditioning on credit adds no forward edge


## Placebo — circular-shift the risk-on labels (1,000 draws)

In [4]:
print(f"observed {R['placebo_obs']:+.1f} bps vs placebo mean {R['placebo_mean']:+.1f} "
      f"(sd {R['placebo_sd']:.1f}) -> {R['placebo_sigma']:+.1f}sigma, p = {R['placebo_p']:.3f}")
print('the observed difference sits INSIDE the shuffle cloud -> nothing to explain')

observed -7.5 bps vs placebo mean -0.2 (sd 16.2) -> -0.4sigma, p = 0.663
the observed difference sits INSIDE the shuffle cloud -> nothing to explain


## Robustness — two eras (split 2017-01-01), regression slope

In [5]:
print(f"2007-2016 (n={R['era_early_n']}): slope {R['era_early_beta']:+.4f}  NW t = {R['era_early_t']:+.2f}")
print(f"2017-2026 (n={R['era_late_n']}): slope {R['era_late_beta']:+.4f}  NW t = {R['era_late_t']:+.2f}")
print('the slope is WRONG-SIGNED in both halves and never clears |t|>=2 -> a weak, persistent anti-lead, not a signal')

2007-2016 (n=501): slope -0.0439  NW t = -1.23
2017-2026 (n=495): slope -0.0910  NW t = -1.33
the slope is WRONG-SIGNED in both halves and never clears |t|>=2 -> a weak, persistent anti-lead, not a signal


## The overlay — SPY when trend>0 else IEF, weekly, costed, vs 100%-SPY buy-and-hold

A switch turns the book over (2 legs); long-only, no borrow. 218 switches over 19 years.

In [6]:
for tag,ns,cg,ac,at in [('1 bp',R['t1_net_sharpe'],R['t1_net_cagr'],R['t1_active_bps'],R['t1_active_t']),
                        ('5 bps',R['t5_net_sharpe'],R['t5_net_cagr'],R['t5_active_bps'],R['t5_active_t'])]:
    print(f"{tag:>5}/leg: net Sharpe {ns:.3f} (B&H {R['t1_bh_sharpe']:.3f}) | net CAGR {cg:+.2f}%% "
          f"(B&H {R['t1_bh_cagr']:+.2f}%%) | active {ac:+.1f} bps/wk (NW t={at:+.2f})")
print(f"drawdown: overlay {R['t1_net_dd']:.0f}%% vs B&H {R['t1_bh_dd']:.0f}%% (the ONLY real benefit)")

 1 bp/leg: net Sharpe 0.644 (B&H 0.645) | net CAGR +7.35%% (B&H +10.61%%) | active -7.6 bps/wk (NW t=-1.22)
5 bps/leg: net Sharpe 0.569 (B&H 0.645) | net CAGR +6.37%% (B&H +10.61%%) | active -9.3 bps/wk (NW t=-1.49)
drawdown: overlay -33%% vs B&H -55%% (the ONLY real benefit)


## Synthetic positive control — the machinery is unbiased

Live: the detector must NOT fire on the null and must recover a planted one-week lead of credit over equity. No network.

In [7]:
import os, sys
sys.path.insert(0, os.path.abspath('..'))
sys.path.insert(0, os.path.abspath(os.path.join('..','..','..')))
import numpy as np
from credit_lead import data, strategy as st
null_t = np.array([st.leadlag_regression(data.synthetic_panel(edge=0.0, seed=865+s, n_days=3000), 4)['beta_t_nw'] for s in range(8)])
print(f"null (edge=0), 8 seeds: regression NW t mean {null_t.mean():+.2f} (sd {null_t.std(ddof=1):.2f}), |t|>=2 in {(abs(null_t)>=2).sum()}/8")
planted = st.synthetic_detect(data.synthetic_panel(edge=0.02, seed=865, n_days=3000), 4)
print(f"planted (edge=0.02): regression NW t = {planted['beta_t_nw']:+.2f}, per-sigma {planted['per_sd_bps']:+.0f} bps, "
      f"active NW t = {planted['active_t_nw']:+.2f}, overlay Sharpe {planted['net_sharpe']:+.2f} vs B&H {planted['bh_sharpe']:+.2f}")

null (edge=0), 8 seeds: regression NW t mean -0.18 (sd 1.06), |t|>=2 in 0/8


planted (edge=0.02): regression NW t = +26.69, per-sigma +988 bps, active NW t = +6.90, overlay Sharpe +3.81 vs B&H +0.19


## Verdict

- **Signal — None.** The credit trend does not lead next-week equity: the predictive-regression slope is wrong-signed and insignificant at every horizon (**-19.6 bps/σ**, NW *t* = **-1.70** at 4 weeks; *t* = -1.58 at 1 week; *t* = -0.23 at 2 weeks), R² never exceeds 0.6%, the risk-on−risk-off difference sits -0.4σ inside a label-shift placebo (p = 0.66), and the wrong sign is consistent across eras. The 20-seed synthetic control recovers a *planted* lead (*t* = +26.7, fires on 0/20 nulls), so the flat real-tape result is a true absence.
- **Tradability — Mirage.** The costed SPY↔IEF overlay never beats buy-and-hold (net Sharpe 0.644 vs 0.645 at 1 bp, active NW *t* = -1.22; it forfeits 3.3%/yr of CAGR) and falls further behind by 5 bps. Its one merit — a -33% vs -55% drawdown — is insurance paid for in forgone return, not an edge.